src/main/scala/com/fulcrumgenomics/umi/VanillaUmiConsensusCaller.scala

```diff
@@ -0,0 +1,359 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.umi
+
+import com.fulcrumgenomics.umi.ConsensusCaller.Base
+import com.fulcrumgenomics.umi.ConsensusCallerOptions._
+import com.fulcrumgenomics.util.NumericTypes._
+import com.fulcrumgenomics.util.{MathUtil, ProgressLogger}
+import htsjdk.samtools._
+import htsjdk.samtools.util.SequenceUtil
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+object ConsensusCallerOptions {
+  type PhredScore = Byte
+
+  /** Various default values for the consensus caller. */
+  val DefaultTag: String                             = "MI"
+  val DefaultErrorRatePreUmi: PhredScore             = 45.toByte
+  val DefaultErrorRatePostUmi: PhredScore            = 40.toByte
+  val DefaultMaxBaseQuality: PhredScore              = 40.toByte
+  val DefaultBaseQualityShift: PhredScore            = 10.toByte
+  val DefaultMinConsensusBaseQuality: PhredScore     = 13.toByte
+  val DefaultMinReads: Int                           = 1
+  val DefaultMinMeanConsensusBaseQuality: PhredScore = 13.toByte
+  val DefaultRequireConsensusForBothPairs: Boolean   = true
+}
+
+/** Holds the parameters/options for consensus calling. */
+case class ConsensusCallerOptions(tag: String                             = DefaultTag,
+                                  errorRatePreUmi: PhredScore             = DefaultErrorRatePreUmi,
```

Comment:
I'm not sure.  I think this class should have been renamed to `VanillaUmiConsensusCallingOptions`, in which case I think preUmi/postUmi makes most sense.  I wanted to pick another name as it's conceivable you have some other way to group reads without UMIs; e.g. you could have clones of a single molecule in a well and then make a library such that the entire library/RG is from one source molecule.